lib/consolidate.js

```diff
@@ -91,13 +93,24 @@ function read(path, options, fn) {
   if (cached) return fn(null, str);
 
   // read
-  fs.readFile(path, 'utf8', function(err, str){
-    if (err) return fn(err);
-    // remove extraneous utf8 BOM marker
-    str = str.replace(/^\uFEFF/, '');
-    if (options.cache) readCache[path] = str;
-    fn(null, str);
-  });
+  if (nowReading[path]) {
+    if (!readListeners[path]) readListeners[path] = [fn];
+    else readListeners[path].push(fn);
+  }else{
+    nowReading[path] = true;
+    fs.readFile(path, 'utf8', function(err, str){
+      if (err) return fn(err);
```

Comment:
Okay, I've verified that this fixes the issue.  Nice pull request by the way.  The only feedback I have is that you may want to call the callbacks in the order that they were issued.  Currently the first callback is being called after all the rest.